src/services/notifier/handlers/provider.ts

```diff
@@ -77,6 +82,29 @@ export const handlers = {
 
     if (subscriptionService.emit) subscriptionService.emit('created', wrapEvent('SubscriptionCreated', subscription))
     logger.info(`Created new Subscription ${hash} by Consumer ${consumer} for Provider ${provider}`)
+  },
+  async FundsWithdrawn (event: FundsWithdrawn, { subscriptionService }: NotifierServices): Promise<void> {
+    const { provider, hash, amount, token } = event.returnValues
+
+    const withdrawnEvent = {
+      provider,
+      hash,
+      amount,
+      token
+    }
+    try {
+      await SubscriptionModel.update(
+        { withdrawableFunds: new BigNumber(0) },
```

Comment:
I agree based on comments in description, this event is probably unnecessary and also it's assuming all funds withdrawn by setting to zero. So, it's probably better to rely on notifier since we are doing that anyway in findOrCreateSubscription

Reply:
Thanks, makes sense. I just removed the last commit.